cpp/src/aztec/honk/composer/standard_honk_composer.test.cpp

```diff
@@ -336,136 +335,81 @@ TEST(standard_honk_composer, test_check_sumcheck_relations_correctness)
-        q_2_univariate.value_at(0) = q_2[i];
-        StandardUnivariate q_3_univariate(0);
-        q_3_univariate.value_at(0) = q_3[i];
-        StandardUnivariate q_c_univariate(0);
-        q_c_univariate.value_at(0) = q_c[i];
-        StandardUnivariate sigma_1_univariate(0);
-        sigma_1_univariate.value_at(0) = sigma_1[i];
-        StandardUnivariate sigma_2_univariate(0);
-        sigma_2_univariate.value_at(0) = sigma_2[i];
-        StandardUnivariate sigma_3_univariate(0);
-        sigma_3_univariate.value_at(0) = sigma_3[i];
-        StandardUnivariate id_1_univariate(0);
-        id_1_univariate.value_at(0) = id_1[i];
-        StandardUnivariate id_2_univariate(0);
-        id_2_univariate.value_at(0) = id_2[i];
-        StandardUnivariate id_3_univariate(0);
-        id_3_univariate.value_at(0) = id_3[i];
-        StandardUnivariate L_first_univariate(0);
-        L_first_univariate.value_at(0) = L_first[i];
-        StandardUnivariate L_last_univariate(0);
-        L_last_univariate.value_at(0) = L_last[i];
-
-        using POLYNOMIAL = bonk::StandardArithmetization::POLYNOMIAL;
-        univariate_array[POLYNOMIAL::W_L] = w_1_univariate;
-        univariate_array[POLYNOMIAL::W_R] = w_2_univariate;
-        univariate_array[POLYNOMIAL::W_O] = w_3_univariate;
-        univariate_array[POLYNOMIAL::Z_PERM] = z_perm_univariate;
-        univariate_array[POLYNOMIAL::Z_PERM_SHIFT] = z_perm_shift_univariate;
-        univariate_array[POLYNOMIAL::Q_M] = q_m_univariate;
-        univariate_array[POLYNOMIAL::Q_L] = q_1_univariate;
-        univariate_array[POLYNOMIAL::Q_R] = q_2_univariate;
-        univariate_array[POLYNOMIAL::Q_O] = q_3_univariate;
-        univariate_array[POLYNOMIAL::Q_C] = q_c_univariate;
-        univariate_array[POLYNOMIAL::SIGMA_1] = sigma_1_univariate;
-        univariate_array[POLYNOMIAL::SIGMA_2] = sigma_2_univariate;
-        univariate_array[POLYNOMIAL::SIGMA_3] = sigma_3_univariate;
-        univariate_array[POLYNOMIAL::ID_1] = id_1_univariate;
-        univariate_array[POLYNOMIAL::ID_2] = id_2_univariate;
-        univariate_array[POLYNOMIAL::ID_3] = id_3_univariate;
-        univariate_array[POLYNOMIAL::LAGRANGE_FIRST] = L_first_univariate;
-        univariate_array[POLYNOMIAL::LAGRANGE_LAST] = L_last_univariate;
-
-        sumcheck_typed_polynomial_vector.push_back(univariate_array);
-    }
-    // Check all relations at all indices
-    for (size_t i = 0; i < prover.key->circuit_size; i++) {
-        round.accumulate_relation_univariates_testing(
-            sumcheck_typed_polynomial_vector[i], results, { beta, gamma, public_input_delta });
-        EXPECT_EQ(std::get<0>(results), ArithmeticUnivariate(0));
-        EXPECT_EQ(std::get<1>(results), GrandProductComputationUnivariate(0));
-        EXPECT_EQ(std::get<2>(results), GrandProductInitializationUnivariate(0));
+        // // Compute an array containing all the evaluations at a given row i
+        std::array<fr, num_polynomials> transposed;
```

Comment:
Isn't this operation what you created `transposed_univariate_array_at` for?

Reply:
The problem was that that function only worked over `Univariate` since it doesn't have the `operator[]` defined. I moved it to the test it was being used it. In retrospect it may have been a bit too generic and the following loop is simple enough.